bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/discovery/BridgeDiscoveryParticipant.java

```diff
@@ -89,73 +101,159 @@ public String getServiceType() {
         return "_http._tcp.local.";
     }
 
+    /**
+     * This method is frequently called by the mDNS discovery framework in different
+     * threads with individual service info instances.
+     * <p>
+     * Different service info objects can refer to the same Bosch SHC controller,
+     * e.g. the controller may be reachable via a <code>192.168.*.*</code> IP and an
+     * IP in the <code>169.254.*.*</code> range. The response from the controller
+     * contains the actual resolved IP address.
+     * <p>
+     * We ignore mDNS events if they do not contain any IP addresses or if the name
+     * property does not start with <code>Bosch SHC</code>.
+     */
     @Override
     public @Nullable DiscoveryResult createResult(ServiceInfo serviceInfo) {
-        logger.trace("Bridge Discovery started for {}", serviceInfo);
+        if (logger.isTraceEnabled()) {
+            logger.trace("Bridge discovery invoked with mDNS service info {}", serviceInfo);
+        }
+
+        String name = serviceInfo.getName();
+        if (name == null || !name.startsWith(NAME_PREFIX_BOSCH_SHC)) {
```

Comment:
This seems to contradict https://github.com/openhab/openhab-addons/pull/15817#discussion_r1380824248. Does it mean that the addon metadata can now be extended to support suggestion finder?

Reply:
I don't know to be honest 😉 I merely reported what I saw in the mDNS app. But debugging the events now I can see that we get qualified names like `Bosch SHC [64-da-a0-xx-xx-xx]._94eb080a._sub._http._tcp.local.`. The components are:
* name: `Bosch SHC [64-da-a0-xx-xx-xx]`, where the square brackets contain the bridge MAC address
* subtype: `_94eb080a`
* type: `_http._tcp.local.` (application, protocol and domain)

If I understand correctly, the discovery participant can only listen to a certain `type` of events. I don't know if the suggestion finder supports more advanced filtering. But it would have to support some kind of "starts-with" filter since the MAC addresses are specific to the controller. I don't think that the `subtype` will be of use because other devices (such as my printer) have the same subtype. Maybe @andrewfg can help us here?

Reply:
> Maybe @andrewfg can help us here?

The suggestion finder uses a regex match where you can define any match pattern.

Reply:
Cool 😎 Am I right to assume the regex is applied to the full qualified name and not just the `type` part? If so, should I create a pull request or do you want to create one since you know better where to add this and which regexes are to be used?

Reply:
MDNS results return fqn and optional parameters that are service specific. The regex can match on any of those.

Reply:
@david-pace probably you just need to add the following to your addon.xml file..

```	<discovery-methods>
		<discovery-method>
			<service-type>mdns</service-type>
			<discovery-parameters>
				<discovery-parameter>
					<name>mdnsServiceType</name>
					<value>_http._tcp.local.</value>
				</discovery-parameter>
			</discovery-parameters>
			<match-properties>
				<match-property>
					<name>name</name>
					<regex>Bosch SHC.*</regex>
				</match-property>
			</match-properties>
		</discovery-method>
	</discovery-methods>
```

Reply:
Now included in this pull request. Thank you very much @andrewfg ❤️